ksql-engine/src/main/java/io/confluent/ksql/security/KsqlAccessValidatorProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.security;
+
+import io.confluent.ksql.exception.KsqlTopicAuthorizationException;
+import java.util.Collections;
+import java.util.Set;
+import org.apache.kafka.common.acl.AclOperation;
+
+/**
+ * An implementation of {@link KsqlAccessValidator} that provides authorization checks
+ * from the Kafka service.
+ */
+public class KsqlAccessValidatorProvider implements KsqlAccessValidator {
```

Comment:
could we rename this to `KsqlDefaultAccessValidator` or `KsqlBaseAccessValidator` to keep the type hierarchy more clearer to read.. Provider typically handles the functionality that the new `KsqlAccessValidator` takes up

Reply:
I used `KsqlBackendAccessValidator` to interpret that this implementation will check access on the backend service. This service might be Kafka or SR (future impl).

I did not use `Default...` because we can have a different access validator based on RBAC services instead of the backend service.